asset/asset.go

```diff
@@ -727,14 +742,11 @@ func NewScriptKeyBip86(rawKey keychain.KeyDescriptor) ScriptKey {
 // GenesisSigner is used to sign the assetID using the group key public key
 // for a given asset.
 type GenesisSigner interface {
-	// SignGenesis tweaks the public key identified by the passed key
-	// descriptor with the the first passed Genesis description, and signs
-	// the second passed Genesis description with the tweaked public key.
-	// For minting the first asset in a group, only one Genesis object is
-	// needed, since we tweak with and sign over the same Genesis object.
-	// The final tweaked public key and the signature are returned.
-	SignGenesis(keychain.KeyDescriptor, Genesis,
-		*Genesis) (*btcec.PublicKey, *schnorr.Signature, error)
+	// SignVirtualTx generates a signature according to the passed signing
+	// descriptor and TX.
+	SignVirtualTx(signDesc *lndclient.SignDescriptor, tx *wire.MsgTx,
```

Comment:
For our purposes, I think we can assume it's just keyspends for now, so we can actually keep the existing `KeyDescriptor`? Still getting thru the diff though.

Reply:
Alternatively, we can rely on this method to produce the virtual tx itself. So it just needs the `Genesis` for the "root" asset, then the asset being created (`asset.Asset`), then it returns the witness as is. Also I don't think the first return value is valid any longer with the new scheme.

Reply:
Resolved offline.